cpp/src/io/json/json_gpu.cu

```diff
@@ -435,63 +450,68 @@ __device__ __inline__ bool is_like_float(
  * @param[out] num_valid_fields The numbers of valid fields in columns
  *
  * @return void
- **/
-__global__ void convert_json_to_columns_kernel(const char *data,
+ */
+__global__ void convert_data_to_columns_kernel(const char *data,
                                                size_t data_size,
                                                const uint64_t *rec_starts,
                                                cudf::size_type num_records,
                                                const data_type *dtypes,
                                                ParseOptions opts,
+                                               bool are_rows_objects,
+                                               col_map_type col_map,
```

Comment:
Yeah, I'm not happy with that part either. I was planning to make a follow-up PR to improve the parameters for all these kernels, since the scope of the parameter issue is beyond this PR. I feel like the proper fix would touch a lot of code unrelated to object parsing. 
@codereport are you okay with this being addressed in a separate PR?

Reply:
As discussed in today's stand up meeting - pushing to follow up PR is fine. Can you set up an issue? Feel free to resolve this conversation afterwards (maybe link the issue here first).

Reply:
Filed https://github.com/rapidsai/cudf/issues/5710